src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,58 @@
+package duke.command;
+
+import java.util.ArrayList;
+
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.task.Task;
+
+/**
+ * Encapsulation of the command of searching for a keyword in the list of tasks.
+ *
+ * @author Sun Ruoxin
+ */
+public class FindCommand extends Command {
+    /** The keyword to be searched. */
+    protected String keyword;
+
+    /**
+     * Class constructor.
+     *
+     * @param keyword the keyword to be searched for
+     */
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    /**
+     * Executes the command.
+     * Searches for the keyword in the list of tasks.
+     * Gives feedback to the user.
+     *
+     * @param tasks the list of tasks
+     * @param ui the UI
+     * @param storage the storage
+     */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
+        ArrayList<Task> matches = tasks.find(keyword);
+//        ui.findMessage(matches);
```

Comment:
If I remember correctly, all comments should be indented based on the line of code below it, so the double slash should be right next to `ui.findMessage(matches);`